src/extension.ts

```diff
@@ -98,14 +99,37 @@ class DocumentationViewProvider implements vscode.WebviewViewProvider{
     // function queries server, and returns html
     async indexServer(words: string) {
         const url = "https://reactdocs-server-1.vercel.app/api/";
-        const body = {
-            "name": words
-        };
-        const resp: { data: ReferenceResponse } = await axios.post(url, body);
-        if (resp.data.success) {
-            return this.buildHtml(resp.data.data.link, resp.data.data.docstring);
+        
+        var docFile = `${__dirname}/docs.json`;
+        // await fsPromises.unlink(docFile);
+
+        // Create file if doesn't exist
+        if(!existsSync(docFile)){
+            const resp: { data: ReferenceResponse } = await axios.get(url);
+            if (resp.data.success) {
+                console.log("Creating file locally!");
+                const file = await fsPromises.writeFile(docFile, JSON.stringify(resp.data.data), 'utf-8');
+            } else {
+                return this.loadWebviewHtml;
+            }
         }
-        return "";
+        const docData = await this.readFile(docFile, words);
+
+        if (docData.length) {
+            return this.buildHtml(docData[0], docData[1]);
+        }
+        return this.loadWebviewHtml;
+    }
+
+    async readFile(docFile: string, words: string) {
```

Comment:
NP: rename `words` to something more specific (maybe `functionName` or `highlightedContent`)